### PR TITLE
[Debug] Fix add new user and customise the environment

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🎯 Goal

Fix the Add User flow used in debugging. Adding a way to provide a custom ApiKey and Token.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: DemoApp/Sources/Components/AppState.swift [hunk 1]
```diff
@@ -20,10 +20,22 @@ final class AppState: ObservableObject {
 
     // MARK: Published
 
-    @Published var apiKey: String = ""
+    @Published var apiKey: String = "" {
+        didSet {
+            print("")
```
Comment (laevandus) [inline DemoApp/Sources/Components/AppState.swift:25]: Remove?

File: DemoApp/Sources/Components/AppEnvironment.swift [hunk 2]
```diff
@@ -33,13 +33,28 @@ extension AppEnvironment {
 
 extension AppEnvironment {
 
-    enum BaseURL: String, Debuggable, CaseIterable {
-        case pronto = "https://pronto.getstream.io"
-        case staging = "https://staging.getstream.io"
-        case demo = "https://getstream.io"
-        case legacy = "https://stream-calls-dogfood.vercel.app"
+    indirect enum BaseURL: Debuggable {
```
Comment (martinmitrevski) [inline DemoApp/Sources/Components/AppEnvironment.swift:36]: ProntoStaging should be added (just merge `main`)

File: DemoApp/Sources/Components/AppState.swift (hunk 1 above)
Comment (martinmitrevski) [reply DemoApp/Sources/Components/AppState.swift:25]: yes, should be removed

**Review by martinmitrevski — COMMENTED** [review]
Looks good, we just need to include pronto staging before merging

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**